sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -163,27 +165,76 @@ AccessToken TokenCredentialImpl::ParseToken(
 
     AccessToken accessToken;
     accessToken.Token = parsedJson[accessTokenPropertyName].get<std::string>();
+    accessToken.ExpiresOn = std::chrono::system_clock::now();
 
     if (parsedJson.contains(expiresInPropertyName))
     {
-      accessToken.ExpiresOn
-          = std::chrono::system_clock::now()
-          + std::chrono::seconds(
-                parsedJson[expiresInPropertyName].get<std::chrono::seconds::duration::rep>());
+      try
+      {
+        // 'expires_in' as number (seconds until expiration)
+        accessToken.ExpiresOn += std::chrono::seconds(
+            parsedJson[expiresInPropertyName].get<std::chrono::seconds::duration::rep>());
+
+        return accessToken;
+      }
+      catch (std::exception const&)
+      {
+        try
+        {
+          // 'expires_in' as numeric string (seconds until expiration)
+          accessToken.ExpiresOn += std::chrono::seconds(
+              std::stoi(parsedJson[expiresInPropertyName].get<std::string>()));
+
+          return accessToken;
+        }
+        catch (std::exception const&)
+        {
+          if (expiresOnPropertyName.empty())
+          {
+            throw;
+          }
+        }
+      }
     }
     else if (expiresOnPropertyName.empty())
     {
       ThrowMissingJsonPropertyError(expiresInPropertyName);
     }
-    else if (!parsedJson.contains(expiresOnPropertyName))
+
+    if (!parsedJson.contains(expiresOnPropertyName))
     {
       ThrowMissingJsonPropertyError(expiresOnPropertyName);
     }
-    else
+
+    try
+    {
+      // 'expires_on' as RFC3339 date string (absolute timestamp)
+      accessToken.ExpiresOn = DateTime::Parse(
+          parsedJson[expiresOnPropertyName].get<std::string>(), DateTime::DateFormat::Rfc3339);
+    }
+    catch (std::exception const&)
     {
-      accessToken.ExpiresOn = Azure::DateTime::Parse(
-          parsedJson[expiresOnPropertyName].get<std::string>(),
-          Azure::DateTime::DateFormat::Rfc3339);
+      try
+      {
+        // 'expires_on' as number (posix time representing an absolute timestamp)
```

Comment:
Similar comment to the above.  It seems weird to see these nested try/catch statements, can't you check the values (and the type of the value) instead of depending on the JSON parser to throw?